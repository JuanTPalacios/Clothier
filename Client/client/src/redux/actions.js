const actions = {
  getItems : (items) => ({
    type: 'getItems',
    payload: items
  }),
};

export default actions;