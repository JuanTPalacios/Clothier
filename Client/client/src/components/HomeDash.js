import React from 'react'
import SearchResults from './SearchResults'
import { Link } from "react-router-dom";
import { connect } from 'react-redux';
import actions from '../redux/actions';


function HomeDash({searchToggle, setSearchToggle, items}) {

  const mock = {
    title: 'testItem',
    category: 'jeans',
    image: "https://www.zappos.com/images/z/5/1/4/2/0/9/5142092-p-DETAILED.jpg"
  }

  if(searchToggle) return (
    <SearchResults setSearchToggle={setSearchToggle}></SearchResults>
  )
  return (
    <>
    <div className="categoryScroll">
      Categories:
    </div>
    <div className="dashboardItems">
      {items.title}
      <Link to="/itemDetail">
        <img src={items.image} alt="n/a" />
      </Link>
    </div>
    </>
  )
}

const mapStateToProps = ({items}) => {
  return {
    items: items.items[0],
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(HomeDash);

