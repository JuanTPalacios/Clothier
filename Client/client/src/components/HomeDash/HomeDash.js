import React from 'react'
import SearchResults from'../SearchResults/SearchResults'
import { Link } from "react-router-dom";
import { connect } from 'react-redux';
import actions from '../../redux/actions';
import { useState } from 'react';
import '../../styles/app.css';

const catArr = ['Pants', 'Outerwear Pants and Sets', 'Coats & Outerwear', 'Hoodies & Sweatshirts', 'Sweaters', 'Shirts & Tops', 'Underwear & Intimates', 'Dresses', 'Jeans', 'Socks', 'Skirts', "Kids' Sets"]
const initialState = catArr.map(category => {return {category: category, isActive: ''}})

function HomeDash({items, searchVal, setSelectedItem}) {
  
  const [all, setAll] = useState({category: 'all', isActive: 'is-active'});
  const [filter, setFilter] = useState('all');
  const [categories, setCategories] = useState(initialState);
  const [prevIndex, setPrevIndex] = useState(null)

  function handleClick(cat, index) {
    setAll({...all, isActive:''})
    setFilter(cat.category);
    const newActive= [...categories];
    setPrevIndex(index);
    if (prevIndex !== null) newActive[prevIndex].isActive = '';
    newActive[index].isActive = 'is-active';
    setCategories(newActive);
  }
  
  function handleAllClick() {
    setFilter(all.category);
    setAll({...all, isActive:'is-active'})
    const newActive= [...categories];
    if (prevIndex !== null) newActive[prevIndex].isActive = '';
    setCategories(newActive);
  }


  if(!items) return (
    <div className="loading">Loading...</div>
    )

  const filteredItems = filter === 'all' ? items : items.filter((item) => item.category === filter)
  const length = filteredItems.length;
  const quarter = length===1 ? 4 : Math.floor(length/4);
  return (
    <div>
      {searchVal && 
      <div className="search">
      <SearchResults></SearchResults>
      </div>
      }
      <div className="categories tabs">
        <ul>
          <li className={all.isActive} onClick={() => handleAllClick()}> <a>All</a> </li>
          {categories.map((category, index) =>
            <li className={category.isActive} key={index} onClick={() => handleClick(category, index)} ><a>{category.category}</a></li>
            )}
        </ul>
      </div>
      <div className="dashboardItems tile is-ancestor">
        <div className="tile is-3 is-vertical pt-2">
          {filteredItems.slice(0, quarter).map(item =>
            <div className='tile is-child box item-box' key={item.primaryKey}>
              <Link to={`/itemDetail/${item.primaryKey}`}>
                <img src={item.image} onClick={() => setSelectedItem(actions.getSingleItem(item))} alt="n/a"/>
              </Link>
        </div>
        )}
        </div>
        <div className="tile is-3 is-vertical">
          {filteredItems.slice(quarter, quarter*2).map(item =>
            <div className='tile is-child box item-box' key={item.primaryKey}>
              <Link to={`/itemDetail/${item.primaryKey}`}>
                <img src={item.image} onClick={() => setSelectedItem(actions.getSingleItem(item))} alt="n/a"/>
              </Link>
            </div>
          )}
        </div>
        <div className="tile is-3 is-vertical pt-3">
          {filteredItems.slice((quarter*2), quarter*3).map(item =>
              <div className='tile is-child box item-box' key={item.primaryKey}>
                <Link to={`/itemDetail/${item.primaryKey}`}>
                  <img src={item.image} onClick={() => setSelectedItem(actions.getSingleItem(item))} alt="n/a"/>
                </Link>
              </div>
            )}
        </div>
        <div className="tile is-3 is-vertical">
          {filteredItems.slice((quarter*3)).map(item =>
              <div className='tile is-child box item-box' key={item.primaryKey}>
                <Link to={`/itemDetail/${item.primaryKey}`}>
                  <img src={item.image} onClick={() => setSelectedItem(actions.getSingleItem(item))} alt="n/a"/>
                </Link>
              </div>
            )}
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = ({store}) => {
  return {
    items: store.items,
    searchVal: store.searchVal,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setSelectedItem: (action) => dispatch(action)
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(HomeDash);

