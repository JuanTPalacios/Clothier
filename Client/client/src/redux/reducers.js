import { combineReducers } from 'redux';

const initialState = {
  items: [] ,
  setAuthenticated: false
}

const items = (state = initialState, action) => {
  switch (action.type) {
    case 'getItems':
    return {items: [...state.items, ...action.payload]};
  default:
    return state;
  }
};

// Combining both reducers
const reducers = combineReducers({
  items,
});

export default reducers;