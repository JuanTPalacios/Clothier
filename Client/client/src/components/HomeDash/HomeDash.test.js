import {render} from '@testing-library/react';
import { createStore } from 'redux';

import HomeDash from './HomeDash';
import { Provider } from 'react-redux';
import reducers from '../../redux/reducers';

describe('Home component', () => {
  test('should match the snapshot', () => {
    let store = createStore(reducers, window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__());
    
    const Wrapper = ({ children }) => (
        // you could just use your normal Redux store or create one just for the test
        <Provider store={store}>{children}</Provider>
    );
    const { container } = render(<HomeDash />, {wrapper: Wrapper});
    expect(container.firstChild).toMatchSnapShot();
  })
})