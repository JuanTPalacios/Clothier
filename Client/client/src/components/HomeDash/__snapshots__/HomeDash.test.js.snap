// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match the snapshot 1`] = `
<div>
  
  <div
    class="categories tabs"
  >
    <ul>
      <li
        class="is-active"
      >
         
        <a>
          All
        </a>
         
      </li>
      <li
        class=""
      >
        <a>
          Pants
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Outerwear Pants and Sets
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Coats & Outerwear
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Hoodies & Sweatshirts
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Sweaters
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Shirts & Tops
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Underwear & Intimates
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Dresses
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Jeans
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Socks
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Skirts
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Kids' Sets
        </a>
      </li>
    </ul>
  </div>
  <div
    class="dashboardItems tile is-ancestor"
  >
    <div
      class="tile is-3 is-vertical pt-2"
    />
    <div
      class="tile is-3 is-vertical"
    />
    <div
      class="tile is-3 is-vertical pt-3"
    />
    <div
      class="tile is-3 is-vertical"
    />
  </div>
</div>
`;
